server/src/test/java/io/druid/segment/realtime/appenderator/DefaultOfflineAppenderatorFactoryTest.java

```diff
@@ -0,0 +1,155 @@
+                                           @Override
+                                           public String getFormatString()
+                                           {
+                                             return "processing-%s";
+                                           }
+
+                                           @Override
+                                           public int intermediateComputeSizeBytes()
+                                           {
+                                             return 100 * 1024 * 1024;
+                                           }
+
+                                           @Override
+                                           public int getNumThreads()
+                                           {
+                                             return 1;
+                                           }
+
+                                           @Override
+                                           public int columnCacheSizeBytes()
+                                           {
+                                             return 25 * 1024 * 1024;
+                                           }
+                                         }
+                                     );
+                                     binder.bind(ColumnConfig.class).to(DruidProcessingConfig.class);
+                                   }
+                                 }
+        )
+    );
+    DefaultOfflineAppenderatorFactory defaultOfflineAppenderatorFactory = injector.getInstance(
+        DefaultOfflineAppenderatorFactory.class);
+    ObjectMapper objectMapper = injector.getInstance(ObjectMapper.class);
+    final Map<String, Object> parserMap = objectMapper.convertValue(
+        new MapInputRowParser(
+            new JSONParseSpec(
+                new TimestampSpec("ts", "auto", null),
+                new DimensionsSpec(null, null, null),
+                null,
+                null
+            )
+        ),
+        Map.class
+    );
+    DataSchema schema = new DataSchema(
+        "dataSourceName",
+        parserMap,
+        new AggregatorFactory[]{
+            new CountAggregatorFactory("count"),
+            new LongSumAggregatorFactory("met", "met")
+        },
+        new UniformGranularitySpec(Granularity.MINUTE, QueryGranularities.NONE, null),
+        objectMapper
+    );
+
+    RealtimeTuningConfig tuningConfig = new RealtimeTuningConfig(
+        75000,
+        null,
+        null,
+        temporaryFolder.newFolder(),
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        0,
+        0,
+        null,
+        null
+    );
+
+    Appenderator appenderator = defaultOfflineAppenderatorFactory.build(
+        schema,
+        tuningConfig,
+        new FireDepartmentMetrics()
+    );
+    Assert.assertEquals(null, appenderator.startJob());
+    Assert.assertEquals("dataSourceName", appenderator.getDataSource());
```

Comment:
Maybe try to add some sample row to Appenderator, to see that it works (and cache is non-null)?